p2p/module.go

```diff
@@ -1,119 +1,149 @@
 package p2p
 
 import (
-	"log"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/libp2p/go-libp2p"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	libp2pHost "github.com/libp2p/go-libp2p/core/host"
+	libp2pNetwork "github.com/libp2p/go-libp2p/core/network"
+	"github.com/multiformats/go-multiaddr"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 
+	"github.com/pokt-network/pocket/libp2p/network"
 	"github.com/pokt-network/pocket/logger"
+	"github.com/pokt-network/pocket/p2p/protocol"
 	"github.com/pokt-network/pocket/p2p/providers"
 	"github.com/pokt-network/pocket/p2p/providers/current_height_provider"
 	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
 	persABP "github.com/pokt-network/pocket/p2p/providers/peerstore_provider/persistence"
 	"github.com/pokt-network/pocket/p2p/raintree"
-	"github.com/pokt-network/pocket/p2p/stdnetwork"
-	"github.com/pokt-network/pocket/p2p/transport"
 	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/p2p/utils"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/configs/types"
 	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
 	"github.com/pokt-network/pocket/shared/messaging"
 	"github.com/pokt-network/pocket/shared/modules"
 	"github.com/pokt-network/pocket/shared/modules/base_modules"
 	"github.com/pokt-network/pocket/telemetry"
-	"google.golang.org/protobuf/proto"
-	"google.golang.org/protobuf/types/known/anypb"
 )
 
+const readStreamTimeoutDuration = time.Second * 10
+
+// TECHDEBT: configure timeouts. Consider security exposure vs. real-world conditions).
+// TECHDEBT: parameterize and expose via config.
+// readStreamTimeout is the duration to wait for a read operation on a
+// stream to complete, after which the stream is closed ("timed out").
 var _ modules.P2PModule = &p2pModule{}
 
 type p2pModule struct {
 	base_modules.IntegratableModule
 
-	listener typesP2P.Transport
-	address  cryptoPocket.Address
-
-	logger *modules.Logger
-
-	network typesP2P.Network
-
-	peerstoreProvider     providers.PeerstoreProvider
+	address               cryptoPocket.Address
+	logger                *modules.Logger
+	cfg                   *configs.P2PConfig
+	bootstrapNodes        []string
 	currentHeightProvider providers.CurrentHeightProvider
-
-	bootstrapNodes []string
+	pstoreProvider        providers.PeerstoreProvider
+	identity              libp2p.Option
+	listenAddrs           libp2p.Option
+	// host represents a libp2p network node, it encapsulates a libp2p peerstore
+	// & connection manager. `libp2p.New` configures and starts listening
+	// according to options.
+	// (see: https://pkg.go.dev/github.com/libp2p/go-libp2p#section-readme)
+	host libp2pHost.Host
+	// pubsub is used for broadcast communication
```

Comment:
So, I seem to have refactored out the need for pubsub and just didn't notice such that I came back and cleaned this up. :sweat_smile: We were using it in the libp2p `stdnetwork` analogue but during consolidation its usage reduced to:

```golang
for _, peer := range n.pstore.GetPeerList() {
	if err := utils.Libp2pSendToPeer(n.host, data, peer); err != nil {
		n.logger.Error().
			Err(err).
			Bool("TODO", true).
			Str("pokt address", peer.GetAddress().String()).
			Msg("broadcasting to peer")
		continue
	}
}
```

It would seem what's left over (and we don't need for now) is the module handling the receiving side of the pubsub.

It may make sense to think about this again in terms of pubsub if we get traction with #506 but for now it adds a dependency for no good reason.